os/kernel/log_dump/log_dump.c

```diff
@@ -135,13 +159,16 @@ int log_dump_init(void)
 	is_started_to_save = true;
 
 	return LOG_DUMP_OK;
-}
 
-/* Structures used to compress last partially filled block */
-unsigned char *last_comp_block;
-size_t last_comp_block_size;
-int last_comp_block_ptr;
-bool compress_last_block;
+exit_with_outbuf:
```

Comment:
All failure cases print the same message.
How about printing the log from here?
Then we can reduce the code also.